java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -256,10 +253,24 @@ public MantaClient(final ConfigContext config) {
         this.config = config;
         this.home = ConfigContext.deriveHomeDirectoryFromUser(config.getMantaUser());
 
+        final KeyPair keyPair = new KeyPairFactory(config).createKeyPair();
+
+        final Signer.Builder builder = new Signer.Builder(keyPair);
+        if (ObjectUtils.firstNonNull(
+                config.disableNativeSignatures(),
+                DefaultsConfigContext.DEFAULT_DISABLE_NATIVE_SIGNATURES)) {
+            builder.providerCode("stdlib");
+        }
+        final ThreadLocalSigner signer = new ThreadLocalSigner(builder);
```

Comment:
This block looks the same as in the other constructor.  Should it be DRYed out?

I probably wrote this block,but I admit I am having trouble recalling why `DefaultsConfigContext` needs to explicitly be referenced?  We allow config contexts that don't start with the defaults?

Reply:
People could pass a `StandardConfigContext` which would throw NPEs if accessed directly. I've been fighting with the tradeoff between documenting defaults by directly referencing the `DefaultsConfigContext` and expressing the default behavior through for example structuring if clauses and using `BooleanUtils.isTrue()` for handling nulls. This small check would probably benefit from the latter since it's a simple boolean check though.